### PR TITLE
feat(cli): bail out on missing required web packages

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

Auto installing required web packages doesn't work when it happens in the same process as the bundler starting. This PR changes the behavior to assert, forcing the user to run the install and restart the process -- this is standard in some non-expo web tooling.

# Test Plan

Run `expo start --web` or `expo export:web` in a project without `react-native-web`.

### CODE REVIEW

File: packages/@expo/cli/src/start/doctor/dependencies/__tests__/ensureDependenciesAsync-test.ts [hunk 1]
```diff
@@ -15,12 +15,23 @@ describe(createInstallCommand, () => {
...
           { pkg: '@other/pkg', file: '' },
           { pkg: 'bacon', file: '', version: '~1.0.0' },
         ],
       })
     ).toBe('npm install @other/pkg bacon@~1.0.0');
   });
+  it(`formats pnpm`, () => {
+    expect(
+      createInstallCommand({
+        manager: 'pnpm',
```
Comment (byCedric) [inline packages/@expo/cli/src/start/doctor/dependencies/__tests__/ensureDependenciesAsync-test.ts:29]: thank-you

File: packages/@expo/cli/src/start/doctor/dependencies/ensureDependenciesAsync.ts [hunk 2]
```diff
@@ -77,7 +75,12 @@ export async function ensureDependenciesAsync(
     title = '';
   }
 
-  const installCommand = createInstallCommand({ isYarn, packages: missing });
+  const managerName = PackageManager.resolvePackageManager(projectRoot);
```
Comment (byCedric) [inline packages/@expo/cli/src/start/doctor/dependencies/ensureDependenciesAsync.ts:78]: Just a random thought, but would it help to have a "default" manager returned from this too, or as constant?

File: packages/@expo/cli/src/start/doctor/dependencies/ensureDependenciesAsync.ts [hunk 3]
```diff
@@ -96,18 +99,18 @@ function wrapForTerminal(message: string): string {
...
+  manager: PackageManager.NodePackageManager;
   packages: {
     file: string;
     pkg: string;
     version?: string | undefined;
   }[];
 }) {
   return (
-    (isYarn ? 'yarn add' : 'npm install') +
+    (manager === 'yarn' ? `${manager} add` : `${manager} install`) +
```
Comment (byCedric) [inline packages/@expo/cli/src/start/doctor/dependencies/ensureDependenciesAsync.ts:113]: I think `pnpm` also uses add:
image

**Review by byCedric — APPROVED** [review]
shipit

File: packages/@expo/cli/src/start/doctor/dependencies/ensureDependenciesAsync.ts (hunk 3 above)
Comment (EvanBacon) [reply packages/@expo/cli/src/start/doctor/dependencies/ensureDependenciesAsync.ts:113]: Looks like it also uses `install`

**Pull request merged**